docs/source/pages/tutorials/detect_malicious_package.rst

```diff
@@ -0,0 +1,186 @@
+      check_passed(component_id, "mcn_detect_malicious_metadata_1").
+
+
+  apply_policy_to("check-django", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi/django@.*", purl).
+
+The ``match`` constraint in this policy allows us to apply the policy on all versions of ``django``. The result of this command should show that the policy succeeds with a zero exit code:
+
+.. code-block:: javascript
+
+    passed_policies
+        ['check-django']
+    component_satisfies_policy
+        ['1', 'pkg:pypi/django@5.0.6', 'check-django']
+    failed_policies
+    component_violates_policy
+
+Note that the ``match`` constraint applies a regex and can be expanded to enforce the ``mcn_detect_malicious_metadata_1`` check to pass on all Python packages analyzed so far by Macaron:
+
+.. code-block:: prolog
+
+  apply_policy_to("check-django", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi.*", purl).
+
+Note that if a policy fails to pass, Macaron returns a none-zero error code.
+
+''''''''''''''''''''''''''''''''''
+Analyzing django with dependencies
+''''''''''''''''''''''''''''''''''
+
+Macaron supports analyzing dependencies of a package and runs the same set of checks on the dependencies as the main target package. To analyze the dependencies of ``django@5.0.6`` Python package, you can either :ref:`generate an SBOM <python-sbom>` yourself or :ref:`point Macaron to a virtual environment <python-venv-deps>` where ``django`` is installed.
+
+
+Let's assume ``/tmp/.django_venv`` is the virtual environment where ``django@5.0.6`` is installed. Run Macaron as follows to analyze ``django`` and its dependencies.
+
+.. code-block:: shell
+
+  ./run_macaron.sh analyze -purl pkg:pypi/django@5.0.6 --python-venv "/tmp/.django_venv"
+
+
+By default Macaron only checks the direct dependencies. To turn on recursive dependency analysis, add the following to the ``configurations.ini`` file:
+
+.. code-block:: ini
+
+  [dependency.resolver]
+  recursive = True
+
+And pass that to the ``analyze`` command:
+
+.. code-block:: shell
+
+  ./run_macaron.sh --defaults-path configurations.ini analyze -purl pkg:pypi/django@5.0.6 --python-venv "/tmp/.django_venv"
+
+To learn more about changing configurations see :ref:`here <change-config>`.
+
+Now we can enforce the policy below to ensure that the ``mcn_detect_malicious_metadata_1`` check always passes on ``django`` and its dependencies and none of the dependencies have malicious behavior.
+
+.. code-block:: prolog
+
+  #include "prelude.dl"
+
+  Policy("check-dependencies", component_id, "Check the dependencies of django.") :-
+    is_component(component_id, _),
+    transitive_dependency(component_id, dependency),
+    check_passed(dependency, "mcn_detect_malicious_metadata_1").
+
+  apply_policy_to("check-dependencies", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi.*", purl).
```

Comment:
I think we should match the PURL to `django` instead. If we use `pkg:pypi,*` this policy will be run against all PYPI components which include the direct dependencies. 
If a component doesn't have any dependencies, it will fail the policy, which I don't think is desirable. 
For example, this could be the result of using this policy:
```
component_satisfies_policy
    ['1', 'pkg:pypi/django@5.0.6', 'check-dependencies']
passed_policies
component_violates_policy
    ['2', 'pkg:pypi/asgiref@3.8.1', 'check-dependencies']
    ['3', 'pkg:pypi/sqlparse@0.5.1', 'check-dependencies']
    ['4', 'pkg:pypi/pip@22.3.1', 'check-dependencies']
failed_policies
    ['check-dependencies']
```

Reply:
Yes, this was meant to be matching `django` as you can see in the related integration policy `[policy-recursive-deps.dl](https://github.com/oracle/macaron/blob/24090493071ca4d9e3e5998338935f6410e9f41c/tests/integration/cases/django_with_dep_resolution_virtual_env_as_input/policy-recursive-deps.dl#L6-L9)`.

See commit [728d753](https://github.com/oracle/macaron/pull/816/commits/728d753678ee948b303d9b2fb9c672f245ed7ee0)